src/test/app/Vault_test.cpp

```diff
@@ -3597,7 +3597,32 @@ class Vault_test : public beast::unit_test::suite
         });
 
         testCase(18, [&, this](Env& env, Data d) {
-            testcase("Scale deposit overflow on second deposit");
+            testcase("MPT scale deposit overflow");
+            // The computed number of shares can not be represented as an MPT
+            // without truncation
+
+            {
+                auto tx = d.vault.deposit(
+                    {.depositor = d.depositor,
+                     .id = d.keylet.key,
+                     .amount = d.asset(5)});
+                env(tx, ter{tecPRECISION_LOSS});
+                env.close();
+            }
+        });
+
+        testCase(13, [&, this](Env& env, Data d) {
+            testcase("MPT scale deposit overflow on first deposit");
+            auto tx = d.vault.deposit(
+                {.depositor = d.depositor,
+                 .id = d.keylet.key,
+                 .amount = d.asset(10)});
+            env(tx, ter{tecPRECISION_LOSS});
+            env.close();
+        });
```

Comment:
I guess this test means that `vaultMaximumIOUScale` in `Protocol.h` should be cut down, perhaps to 15 ? Which will require futher changes in `Vault_test.cpp`